packages/pigeon/test/kotlin/proxy_api_test.dart

```diff
@@ -0,0 +1,944 @@
+        expect(
+          collapsedCode,
+          contains(
+            r'api.pigeonRegistrar.instanceManager.addDartCreatedInstance(api.pigeon_defaultConstructor(',
+          ),
+        );
+      });
+
+      test('multiple params constructor', () {
+        final Enum anEnum = Enum(
+          name: 'AnEnum',
+          members: <EnumMember>[EnumMember(name: 'one')],
+        );
+        final Root root = Root(
+          apis: <Api>[
+            AstProxyApi(name: 'Api', constructors: <Constructor>[
+              Constructor(
+                name: 'name',
+                parameters: <Parameter>[
+                  Parameter(
+                    type: const TypeDeclaration(
+                      isNullable: false,
+                      baseName: 'int',
+                    ),
+                    name: 'validType',
+                  ),
+                  Parameter(
+                    type: TypeDeclaration(
+                      isNullable: false,
+                      baseName: 'AnEnum',
+                      associatedEnum: anEnum,
+                    ),
+                    name: 'enumType',
+                  ),
+                  Parameter(
+                    type: const TypeDeclaration(
+                      isNullable: false,
+                      baseName: 'Api2',
+                    ),
+                    name: 'proxyApiType',
+                  ),
+                  Parameter(
+                    type: const TypeDeclaration(
+                      isNullable: true,
+                      baseName: 'int',
+                    ),
+                    name: 'nullableValidType',
+                  ),
+                  Parameter(
+                    type: TypeDeclaration(
+                      isNullable: true,
+                      baseName: 'AnEnum',
+                      associatedEnum: anEnum,
+                    ),
+                    name: 'nullableEnumType',
+                  ),
+                  Parameter(
+                    type: const TypeDeclaration(
+                      isNullable: true,
+                      baseName: 'Api2',
+                    ),
+                    name: 'nullableProxyApiType',
+                  ),
+                ],
+              )
+            ], fields: <ApiField>[], methods: <Method>[]),
+            AstProxyApi(
+              name: 'Api2',
+              constructors: <Constructor>[],
+              fields: <ApiField>[],
+              methods: <Method>[],
+            ),
+          ],
+          classes: <Class>[],
+          enums: <Enum>[anEnum],
+        );
+        final StringBuffer sink = StringBuffer();
+        const KotlinGenerator generator = KotlinGenerator();
+        generator.generate(
+          const KotlinOptions(),
+          root,
+          sink,
+          dartPackageName: DEFAULT_PACKAGE_NAME,
+        );
+        final String code = sink.toString();
+        final String collapsedCode = _collapseNewlineAndIndentation(code);
```

Comment:
If this test is not generated consider adding how someone should update these strings.